File: stepik-union/src/main/java/org/stepik/core/courseFormat/stepHelpers/StringQuizHelper.java
package org.stepik.core.courseFormat.stepHelpers;

import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;
import org.stepik.core.courseFormat.StepNode;

/**
 * @author meanmail
 */
public class StringQuizHelper extends QuizHelper {
    public StringQuizHelper(@NotNull Project project, @NotNull StepNode stepNode) {
        super(project, stepNode);
    }

    @NotNull
    public String getText() {
        initStepOptions();
        return reply.getText();
    }

    public boolean isTextDisabled() {
        initStepOptions();
        return getDataset().isTextDisabled();
    }

    @Override
    public boolean isAutoCreateAttempt() {
        return true;
    }
}

File: stepik-union/src/main/kotlin/org/stepik/core/courseFormat/stepHelpers/StringQuizHelper.kt
package org.stepik.core.courseFormat.stepHelpers

import com.intellij.openapi.project.Project
import org.stepik.core.courseFormat.StepNode


open class StringQuizHelper(project: Project, stepNode: StepNode) : QuizHelper(project, stepNode) {

    val text: String
        get() {
            initStepOptions()
            return reply.text
        }

    val isTextDisabled: Boolean
        get() {
            initStepOptions()
            return dataset.isTextDisabled
        }
}
